src/components/input/input.stories.mdx

```diff
@@ -33,3 +41,64 @@ You can also add `c-input` to `<select>` elements, in which case you’ll get a
     })}
   </Story>
 </Preview>
+
+## Elastic textarea
+
+Add `js-elastic-textarea` to a `<textarea>` element\* to have it expand as the user types more than will fit.
+
+If the `rows` attribute is set, it will be used as the minimum height.
+
+<p>
+  <small>* see also the JavaScript usage notes below</small>
+</p>
+
+<Preview>
+  <Story name="Elastic Textarea" height="350px">
+    {() => {
+      // Use storybook hooks to trigger JS after story renders
+      // @see https://github.com/storybookjs/storybook/issues/7786
+      useEffect(() => {
+        // Select all elastic textarea elements, run create function on each
+        [...document.querySelectorAll('.js-elastic-textarea')].map(
+          (textarea) => {
+            return createElasticTextArea(textarea);
+          }
+        );
+      }, []);
+      return textDemo({
+        type: select(
+          'type',
+          ['text', 'email', 'search', 'date', 'textarea'],
+          'textarea'
+        ),
+        value: text('value', 'a\nb\nc\nd\ne'),
+        placeholder: text('placeholder', 'Placeholder text…'),
+        rows: number('rows', '2'),
+        id: text('id', 'demo-elastic'),
+        class: text('class', 'js-elastic-textarea'),
+      });
+    }}
+  </Story>
+</Preview>
+
+### JavaScript Usage
+
+Once you have added the `js-elastic-textarea` class to your `textarea` element, you'll also need to add some JavaScript that runs after the page loads, like so:
+
+```js
+// Select all elastic textarea elements, run create function on each
+const elasticTextAreas = [
+  ...document.querySelectorAll('.js-elastic-textarea'),
+].map((textarea) => {
+  return createElasticTextArea(textarea);
+});
+```
+
+The `createElasticTextArea` function adds an event listener to the `textarea`, allowing it to respond to user input by growing or shrinking as needed. It returns an object containing a `destroy()` method, if you need to remove the event listener.
```

Comment:
Thoughts about explicitly calling out what event we are using? 🙂 
```suggestion
The `createElasticTextArea` function adds an [`input`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event) event listener to the `textarea`, allowing it to respond to user input by growing or shrinking as needed. It returns an object containing a `destroy()` method, if you need to remove the event listener.
```

Reply:
Do we need to document this implementation detail? I think it does not affect users of this component.

I think it would be better if we only describe only what it does and how to use it, instead of how it works. If someone wants to know how it works, they can look at the source code / comments

Reply:
@calebeby That's fair and a good point. 👍